### PR TITLE
[Publisher][Admin Panel] Add Agency URL and Description Fields to Admin Panel

### LINKED ISSUE #1554: [Publisher] [Admin Panel] Add Agency URL and Description Fields to Admin Panel
In the current agency provisioning section of the admin panel, users should be able to update both an agency's URL and description. These fields can be added in the existing request to the `create_or_update_agency` endpoint as "agency_url" and "agency_description".

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Added Agency URL and Description Fields to Admin Panel

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

closes #1554 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by mxosman — APPROVED** [review]

File: publisher/src/components/AdminPanel/AgencyProvisioning.tsx [hunk 1]
```diff
@@ -294,12 +298,28 @@ export const AgencyProvisioning: React.FC<ProvisioningProps> = observer(
     //   setZipcodeValidationError("Please enter a 5-digit zipcode");
     // };
 
+    const validateAndUpdateURL = (url: string) => {
+      const isValidURL = validateAgencyURL(url);
+      updateAgencyURL(url);
```
[the remark this answers is not in the document]
Comment (nasaownsky) [reply publisher/src/components/AdminPanel/AgencyProvisioning.tsx]: This totally makes sense!

**Comment by nichelle-hall** [comment]
Hi @nasaownsky - The BE has been merged. Feel free to merge the PR.

**Review by mxosman — CHANGES_REQUESTED** [review]
Actually @nasaownsky - there's a bug when you try to delete those fields in the admin panel that seems unrelated to the latest commit. Can you take another look please?

**Comment by nasaownsky** [comment]
>there's a bug when you try to delete those fields in the admin panel that seems unrelated to the latest commit. Can you take another look please? You should be able to delete those fields entirely.

Oh, you're right, seems like it's the old bug, because it is also affects `Name` field. It is not critical for it though since it's required field, but I can see the edge case when someone wants to totally rename an agency and can't entirely delete its name, so I fix it as well. Please check this changes!

File: publisher/src/components/AdminPanel/AgencyProvisioning.tsx [hunk 2]
```diff
@@ -319,10 +322,10 @@ export const AgencyProvisioning: React.FC<ProvisioningProps> = observer(
       saveAgencyName(agencyProvisioningUpdates.name);
 
       // Update final agency description
-      updateAgencyDescription(agencyProvisioningUpdates.agency_description);
+      updateAgencyDescription(descriptionValue);
 
       // Update final agency URL
-      updateAgencyURL(agencyProvisioningUpdates.agency_url);
+      updateAgencyURL(URLValue);
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/AgencyProvisioning.tsx:331]: Wonderful idea to send it all at the end!

**Review by mxosman — APPROVED** [review]
> > there's a bug when you try to delete those fields in the admin panel that seems unrelated to the latest commit. Can you take another look please? You should be able to delete those fields entirely.
> 
> Oh, you're right, seems like it's the old bug, because it is also affects `Name` field. It is not critical for it though since it's required field, but I can see the edge case when someone wants to totally rename an agency and can't entirely delete its name, so I fix it as well. Please check this changes!

Ah, good catch! Super vaguely recall the name field one might've been intentional before we implemented blocking the saving to make sure folks couldn't save with an empty name field - but I might be misremembering. Thank you so much for fixing both!! Truly appreciate it. Looks great to me!

**Comment by nasaownsky** [comment]
Thank you so much, @mxosman!

**Pull request merged**